File: 1539 Kth Missing Positive Number/kth_missing_positive_number.cpp
/*
BY RUSHABH GALA
rushabhvg
GitHub :
https://github.com/rushabhvg
StackOverflow :
https://stackoverflow.com/users/16571212/rushabhvg
LinkedIn:
https://www.linkedin.com/in/rushabhvg/
*/

class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        vector<int> cnt (2002, 0);
        for(int i=0; i<arr.size(); i++) {
            cnt[arr[i]]++;
        }
        for(int i=1; i<2002; i++) {
            if(k>0) {
                if(cnt[i]==0) {
                    k--;
                }
            } else if(k==0) {
                k=i;
                break;
            }
        }
        return k-1;
    }
};

/*
BY RUSHABH GALA
rushabhvg
GitHub :
https://github.com/rushabhvg
StackOverflow :
https://stackoverflow.com/users/16571212/rushabhvg
LinkedIn:
https://www.linkedin.com/in/rushabhvg/
*/

File: 1646-kth-missing-positive-number/kth-missing-positive-number.java
/*
By Rushabh Gala
rushabhvg
GitHub:
https://github.com/rushabhvg
StackOverflow:
https://stackoverflow.com/users/16571212/rushabhvg
LinkedIn:
https://www.linkedin.com/in/rushabhvg/
*/

class Solution {
    public int findKthPositive(int[] arr, int k) {
        int idx = 0;
        int cnt = 0;
        for(int i:arr) {
            if(i==idx+1) {
                idx++;
            }
            while(idx<i-1) {
                idx++;
                cnt++;
                if(cnt==k) {
                    return idx;
                }
            }
            idx = i;
        }
        while(cnt<k) {
            idx++;
            cnt++;
        }
        return idx;
    }
}

/*
By Rushabh Gala
rushabhvg
GitHub:
https://github.com/rushabhvg
StackOverflow:
https://stackoverflow.com/users/16571212/rushabhvg
LinkedIn:
https://www.linkedin.com/in/rushabhvg/
*/
